apps/dashboard/src/helpers/reports.js

```
@@ -698,71 +699,262 @@ export function getTagColor(supportLevel) {
-      // Replace them with non '-String' versions for
-      //   comparison.
-      percentileScoreKey = 'sprPercentile';
-      percentileScoreDisplayKey = 'sprPercentileString';
-      standardScoreKey = 'sprStandardScore';
-      standardScoreDisplayKey = 'sprStandardScoreString';
-    }
-    rawScoreKey = 'roarScore';
+
+  const taskMapping = SCORE_FIELD_MAPPINGS[taskId];
+  if (!taskMapping || !taskMapping[fieldType]) {
+    return undefined;
   }
-  if (taskId === 'sre') {
-    if (grade < 6) {
-      percentileScoreKey = 'tosrecPercentile';
-      percentileScoreDisplayKey = 'tosrecPercentile';
-      standardScoreKey = 'tosrecSS';
-      standardScoreDisplayKey = 'tosrecSS';
-    } else {
-      percentileScoreKey = 'sprPercentile';
-      percentileScoreDisplayKey = 'sprPercentile';
-      standardScoreKey = 'sprStandardScore';
-      standardScoreDisplayKey = 'sprStandardScore';
-    }
-    rawScoreKey = 'sreScore';
+
+  const fieldMapping = taskMapping[fieldType];
+  const fieldName = isLegacy ? fieldMapping.legacy : fieldMapping.new;
+
+  // Handle function-based field names (grade-dependent)
+  if (typeof fieldName === 'function') {
+    return fieldName(grade);
+  }
+
+  return fieldName;
+}
+
+/**
+ * Safely accesses a score value from a scores object with fallback to legacy field names
+ * @param {Object} scoresObject - The scores object to access
+ * @param {string} taskId - The task identifier
+ * @param {number} grade - The grade level
+ * @param {string} fieldType - The type of field to access
+ * @see ./SCORE_FIELD_MIGRATION_GUIDE.md
+ * @returns {*} The score value or undefined if not found
+ */
+export function getScoreValue(scoresObject, taskId, grade, fieldType) {
+  if (!scoresObject || !taskId || fieldType === undefined) {
+    return undefined;
+  }
+
+  if (!ALLOWED_SCORE_FIELD_TYPES.includes(fieldType)) {
+    throw new Error(`Invalid fieldType. Expected one of ${ALLOWED_SCORE_FIELD_TYPES.join(', ')}, but got ${fieldType}`);
   }
```

Is this check necessary, since it is already handled in `resolveFieldName`? The error handling mechanism in `resolveFieldName` would never get a chance to throw, since this statement would throw first. 

We might be able to instead wrap the fuction logic in a try/catch clause. If `resolveFieldName` throws an error, we would let it bubble up into the catch block and log it, handle it, re-throw, return, or whatever else might be appropriate.